packages/react-native/src/Authenticator/__tests__/Authenticator.spec.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { Text } from 'react-native';
+import { render } from '@testing-library/react-native';
+
+import * as UIReactCoreModule from '@aws-amplify/ui-react-core';
+import {
+  AuthenticatorMachineContext,
+  UseAuthenticator,
+} from '@aws-amplify/ui-react-core';
+
+import { Authenticator } from '..';
+
+jest.mock('@aws-amplify/ui-react-core');
+
+function TestChildren() {
+  return <Text>Test Children</Text>;
+}
+const TestComponent = () => {
+  return null;
+};
+
+const useAuthenticatorInitMachineSpy = jest.spyOn(
+  UIReactCoreModule,
+  'useAuthenticatorInitMachine'
+);
+const useAuthenticatorSpy = jest.spyOn(UIReactCoreModule, 'useAuthenticator');
+const useAuthenticatorRouteSpy = jest.spyOn(
+  UIReactCoreModule,
+  'useAuthenticatorRoute'
+);
+
+type MockSelectorParam =
+  | ((
+      context: AuthenticatorMachineContext
+    ) => AuthenticatorMachineContext[keyof AuthenticatorMachineContext][])
+  | undefined;
+
+describe('Authenticator', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    useAuthenticatorRouteSpy.mockReturnValue({
+      Component: TestComponent as any,
+      props: {} as any,
+    });
+  });
+
+  it('behaves as expected in the happy path', () => {
+    const mockUseAuthenticator = (cb: MockSelectorParam) =>
+      cb?.({
+        route: 'signIn',
+      } as unknown as AuthenticatorMachineContext)[0] as unknown as UseAuthenticator;
+    useAuthenticatorSpy.mockImplementation(mockUseAuthenticator);
+
+    const { toJSON } = render(<Authenticator />);
+
+    expect(useAuthenticatorInitMachineSpy).toHaveBeenCalledTimes(1);
+    expect(useAuthenticatorSpy).toHaveBeenCalledTimes(1);
+    expect(useAuthenticatorSpy).toHaveBeenCalledWith(expect.any(Function));
+
+    expect(toJSON()).toMatchSnapshot();
+  });
+
+  it.each(['authenticated', 'signOut'])(
+    'handles the %s route as expected with children',
+    (route) => {
+      useAuthenticatorSpy.mockImplementation(
+        () => ({ route } as unknown as UseAuthenticator)
+      );
+
+      const { toJSON } = render(
+        <Authenticator>
+          <TestChildren />
+        </Authenticator>
+      );
+
+      expect(toJSON()).toMatchSnapshot();
```

Comment:
Any other assertions we want to add, besides snapshot testing? Snapshots offer no guarantees that appearance or behavior is correct/as expected.

Can we also add some tests for unhappy path or any edge cases?

Reply:
Since we are just checking to make sure that the `children` argument is rendered for these `route` values, think we are okay with the snapshot but curious what other assertions/edge cases you are thinking?

Reply:
I was thinking to give the Child component a testID:
```
<Text testId="testChild">Test Children</Text>
```
and then actually checking that it exists in the tree. snapshot testing is only good for checking if UI changed unexpectadly, it doesn't tell you the child rendered.

Reply:
For the edge cases nothing particular off the top of my head, maybe like if you pass in an invalid component as child, does it blow up? Things liek that.

Reply:
> snapshot testing is only good for checking if UI changed unexpectedly, it doesn't tell you the child rendered.

But the child is what the snapshot contains right?

> maybe like if you pass in an invalid component as child, does it blow up?

Is that a test we should write? Like at that point we are just testing the React Native library itself right?

Reply:
thanks for adding assertions!